extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/sql/HllSketchBaseSqlAggregator.java

```diff
@@ -154,6 +155,15 @@ public Aggregation toDruidAggregation(
       }
 
       if (inputType.is(ValueType.COMPLEX)) {
+        if (!isValidComplexInputType(inputType)) {
+          plannerContext.setPlanningError(
+              "Using APPROX_COUNT_DISTINCT() or enabling approximation with COUNT(DISTINCT) is not supported for"
+              + " column type [%s]. You can disable approximation, use COUNT(DISTINCT %s) and rerun the query.",
+              columnArg.getDruidType(),
+              columnArg.getSimpleExtraction().getColumn()
```

Comment:
there is no gurantee that `columnArg.getSimpleExtraction()` is not null; you could just omit it from the message

...and it would be helpfull for the user meeting with this error to have `PlannerConfig.CTX_KEY_USE_APPROXIMATE_COUNT_DISTINCT` suggested to be enabled

Reply:
Makes sense, made the change.